### ISSUE TITLE
Should the "not necessary" be replaced with "incorrect"?

### ISSUE THREAD

**Issue by CeciAc** [MicrosoftDocs/cpp-docs#1022]
@desowin commented on [Tue May 21 2019](https://github.com/MicrosoftDocs/cpp-docs.pl-pl/issues/5)

In the sentence "The underlying OS file handle is also closed by a call to _close, so it is not necessary to call the Win32 function CloseHandle on the original handle." I find the wording misleading.

Reading this sentence, I understand it, "you don't have to do CloseHandle() but if you do, then nothing wrong happens". However, if you call _close() on the descriptor returned by _open_osfhandle() then both the descriptor and the HANDLE it took ownership of gets closed. Thus you shall not CloseHandle() on the HANDLE that was passed to _open_osfhandle(), as it is already closed. You might essentially close some other HANDLE if the value happened to be reused between the call to _close() and CloseHandle().

---
#### Szczegóły dokumentu

⚠ *Nie edytuj tej sekcji. Jest ona wymagana dla docs.microsoft.com ➟ Dołączanie problemu w usłudze GitHub.*

* ID: a25d8611-6401-b7b6-7df9-00fc9ee3987a
* Version Independent ID: 0a3f89eb-3b0c-7a1e-16c0-f394915c3837
* Content: [_open_osfhandle](https://docs.microsoft.com/pl-pl/cpp/c-runtime-library/reference/open-osfhandle?view=vs-2019#feedback)
* Content Source: [docs/c-runtime-library/reference/open-osfhandle.md](https://github.com/MicrosoftDocs/cpp-docs.pl-pl/blob/live/docs/c-runtime-library/reference/open-osfhandle.md)
* Product: **visual-cpp**
* Technology: **cpp-ucrt**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by CeciAc** [comment]
Hi team, 
I'm moving this here since the user suggestion is about the original documentation
best

**Comment by desowin** [comment]
I reported the original issue from the web interface - I am a bit surprised it went to Polish documentation repository. Not really sure what is the correct place to report this, thus posting as comment here.

The website automatically detected that my browser uses Polish, so it did present me the Polish version. As I strongly prefer to read the software documentation in English, I have clicked the button in upper right corner to read the article in English. Then I have scrolled to the bottom and clicked the button to send feedback about the page, entered the details and submitted - and it went to Polish repository even though the "Przeczytaj w języku angielskim" ("Read in English") button was active.

**Comment by corob-msft** [comment]
@desowin 
Thanks. It's interesting to get the perspective of people who don't read the documentation in translation. It's something we didn't worry about when the documentation was first written. We try to simplify the original English now, both to make it easier to translate, and easier for non-native English speakers to read.

I'll simplify the wording here: "The underlying OS file handle is also closed by a call to _close. Don't call the Win32 function CloseHandle on the original handle." If you wrap resources (such as file handles) in objects so you can use RAII style to guarantee release, this call breaks the symmetry. You must treat it as a special case.

**Comment by corob-msft** [comment]
The updated (English) text is live, so I'm closing this issue.
